core/drivers/crypto/marvell/mrvl_ehsm_cryp.c

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2025, Marvell
+ */
+
+#include <trace.h>
+
+#include "mrvl_ehsm_cryp.h"
+#include "ehsm-aes.h"
+#include "ehsm.h"
+#include <string.h>
+
+static struct mutex ehsm_lock = MUTEX_INITIALIZER;
+
+static struct mrvl_ehsm_engine engine = {
+	.lock = &ehsm_lock,
+	.context_count = 0
+};
+
+void mrvl_ehsm_cryp_lock(void)
+{
+	mutex_lock(engine.lock);
+}
+
+void mrvl_ehsm_cryp_unlock(void)
+{
+	mutex_unlock(engine.lock);
+}
+
+#ifdef CFG_EHSM_CONTEXT_STORE_SUPPORT
+bool mrvl_ehsm_aes_cryp_allowed(void)
```

Comment:
Since `mrvl_ehsm_aes_cryp_allowed()` counter part is `mrvl_ehsm_aes_cryp_release()`, I think it would make sense the content allocation function is called `mrvl_ehsm_aes_cryp_allocate()`.

Or maybe rename both to `mrvl_ehsm_aes_cryp_get()`/`mrvl_ehsm_aes_cryp_put()`.